### PR TITLE
Add and delete card attachments

### PR DESCRIPTION
**Pull request by foundrium**
1. Fixed add implementation from [#14](https://github.com/dillenmeister/Trello.NET/pull/14) comments
2. Added delete capability
3. Add async implementation

### CODE REVIEW

File: TrelloNet/Cards/Card.cs [hunk 1]
```diff
@@ -51,13 +51,19 @@ public class CardBadges
...
         }
 
-        public class Attachment
+        public class Attachment : IAttachmentId
         {
             public string Id { get; set; }
             public string IdMember { get; set; }
             public string Name { get; set; }
             public string Url { get; set; }
+            public string MimeType { get; set; }
```
Comment (dillenmeister) [inline TrelloNet/Cards/Card.cs]: Remove MimeType

File: TrelloNet/Cards/NewAttachment.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+		/// <summary>
+		/// A string with a length from 1 to 16384 (optional).
+		/// </summary>
+		public string Url { get; set; }
+
+		/// <summary>
+		/// A string with a length from 1 to 256 (optional).
+		/// </summary>
+		public string MimeType { get; set; }
```
Comment (dillenmeister) [inline TrelloNet/Cards/NewAttachment.cs]: Remove MimeType

File: TrelloNet/Cards/NewAttachment.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+		/// <param name="name">A string with a length from 1 to 256.</param>
+		/// <param name="url">A string with a length from 1 to 16384.</param>
+		/// <param name="mimeType">A string with a length from 1 to 256.</param>
+		public NewAttachment(string name, string url, string mimeType)
```
Comment (dillenmeister) [inline TrelloNet/Cards/NewAttachment.cs]: Have two constructors:
public `NewAttachment(string url, string name)` and `public NewAttachment(string url)`

Makes sense?

**Comment by dillenmeister** [comment]
Since `MimeType` doesn't seem to work as expected I would like it removed again. Sorry... And I suggest two constructors for the `NewAttachment` class as I mentioned above (if you agree).

With these changes I will happily merge this pull request! Thanks a lot!

(I can make these changes if you want to, but I think it's bad etiquette of me not to ask you to do it first).

BTW, it's possible to add more commits to the same pull request instead of closing it and opening a new one (or did you do a new one because of 24pullrequests?).

**Comment by foundrium** [comment]
Done. Yeah I did it for 24pullrequests since I was doing a little more work than something like removing MimeType :) Hope people find this useful. Its a very nice client.

**Pull request merged**